libraries/rib-base/src/main/java/com/badoo/ribs/routing/state/action/single/Ext.kt

```diff
@@ -0,0 +1,34 @@
+package com.badoo.ribs.routing.state.action.single
+
+import android.os.Parcelable
+import com.badoo.ribs.core.Node
+import com.badoo.ribs.routing.state.RoutingContext
+import com.badoo.ribs.routing.transition.TransitionDirection
+import com.badoo.ribs.routing.transition.TransitionElement
+
+internal fun <C : Parcelable> Node<*>.createTransitionElements(
+    item: RoutingContext.Resolved<C>,
+    direction: TransitionDirection,
+    addedOrRemoved: Boolean
+): List<TransitionElement<C>> {
+    val elements = mutableListOf<TransitionElement<C>>()
+
+    val ribView = view
+    if (ribView != null) {
+        elements += TransitionElement(
+            configuration = item.routing.configuration, // TODO consider passing the whole RoutingElement
+            direction = direction,
+            addedOrRemoved = addedOrRemoved,
+            identifier = identifier,
+            view = ribView.androidView
+        )
+    } else {
+        children.forEach {
```

Comment:
I think we can make the search a little better by checking `isViewless` property of the node.

If the node is not viewless and do not have any view (`!isViewless && ribView == null`), then this node is in a backstack, and there is no need to got deeper through the tree.

Reply:
Also the check can be done in a little different way. Use `isViewless` as primary check and start search for children, otherwise return current `view` if it is not null.

Reply:
`isAttachedToView` or `isActive` can also be used, but have not investigated the logic behind them.